qiskit_optimization/converters/linear_inequality_to_penalty.py

```diff
@@ -0,0 +1,319 @@
+# This code is part of Qiskit.
+#
+# (C) Copyright IBM 2020, 2021.
+#
+# This code is licensed under the Apache License, Version 2.0. You may
+# obtain a copy of this license in the LICENSE.txt file in the root directory
+# of this source tree or at http://www.apache.org/licenses/LICENSE-2.0.
+#
+# Any modifications or derivative works of this code must retain this
+# copyright notice, and modified files need to carry a notice indicating
+# that they have been altered from the originals.
+
+"""Converter to convert a problem with equality constraints to unconstrained with penalty terms."""
+
+import copy
+import itertools
+import logging
+from math import fsum
+from typing import Optional, cast, Union, Tuple, List
+
+import numpy as np
+
+from .quadratic_program_converter import QuadraticProgramConverter
+from ..exceptions import QiskitOptimizationError
+from ..problems.constraint import Constraint, ConstraintSense
+from ..problems.quadratic_objective import QuadraticObjective
+from ..problems.quadratic_program import QuadraticProgram
+from ..problems.variable import Variable
+
+logger = logging.getLogger(__name__)
+
+
+class LinearInequalityToPenalty(QuadraticProgramConverter):
+    r"""Convert a problem of known constraints to unconstrained with penalty terms.
+
+    There are known constraints which do not require to add slack variables to
+    construct penalty terms [1].
+
+    Supported known constraint in this class is shown below.
+
+    .. math::
+      \begin{array}{|l|l|}
+      \hline \text { Classical Constraint } & \text { Equivalent Penalty } \\
+      \hline x+y \leq 1 & P(x y) \\
+      \hline x+y=1 & P(1-x-y+2 x y) \\
```

Comment:
I think this case is already treated as `LinearEqualityToPenalty`. Could you verify it?

Reply:
Document error. Corrected to use x+y>=1.